scripts/discourse-sync/client.py

```diff
@@ -0,0 +1,22 @@
+import pydiscourse
+
+# Client extends pydiscourse.DiscourseClient with new methods
+class Client(pydiscourse.DiscourseClient):
+    def lock_post(self, post_id, locked, **kwargs):
+        """
+        Lock a post from being edited
+        https://docs.discourse.org/#tag/Posts/operation/lockPost
+
+        Args:
+            post_id: integer
+            locked: bool
+        """
+        if bool(locked):
+            kwargs["locked"] = "true"
+        else:
+            kwargs["locked"] = "false"
+        return self._put("posts/{}/locked.json".format(post_id), **kwargs)
+
+    def add_staff_notice(self, post_id, notice, **kwargs):
+        kwargs["notice"] = notice
+        return self._put("posts/{}/notice".format(post_id), **kwargs)
```

Comment:
As above, you can use an f-string:

```suggestion
        return self._put(f"posts/{post_id}/notice", **kwargs)
```

Also as above, could probably simplify it to a single line:

```suggestion
        return self._put(f"posts/{post_id}/notice", notice=notice, **kwargs)
```